### ISSUE TITLE
Downscore clients with SSZ_SNAPPY_ERROR_OVER_SSZ_MAX_SIZE error

### ISSUE THREAD

**Issue by twoeths** [ChainSafe/lodestar#6315]
### Describe the bug

Got a lot of errors like this

```
Jan-17 07:38:21.909[network]       verbose: Req  error method=metadata, version=2, encoding=ssz_snappy, client=Unknown, peer=16...pWVhz9, requestId=403654, code=SSZ_SNAPPY_ERROR_OVER_SSZ_MAX_SIZE, maxSize=17, sszDataLength=24
Error: SSZ_SNAPPY_ERROR_OVER_SSZ_MAX_SIZE
```

that only happens to "Unknown" client, I don't see we downscore them a lot, maybe we need to apply more penalties on them. If it happens a lot it may be disconnected by our peer manager heart beat
<img width="1507" alt="Screenshot 2024-01-17 at 16 15 57" src="https://github.com/ChainSafe/lodestar/assets/10568965/7eb5fe46-9139-4ff4-9b08-2781fc9e3062">

### Expected behavior

I think this error is similar to "REQUEST_ERROR_INVALID_RESPONSE_SSZ", need to apply low tolerance there, i.e. ~5 occurrences will get the peer banned

### Operating system

Linux

### Lodestar version or commit hash

v1.14.0

**Comment by nflaig** [comment]
There is another, possibly related error I am seeing on my Lodestar node (v1.14.0-rc.1)

```
Jan-17 09:17:23.039[network]         error: Error setting agentVersion for the peer peerId=16Uiu2HAmKpgbkNA2AKYvt4qfdwFsSwuM7NwE8aMMHoMJiCBAxKAL - NotFound: 
Error: NotFound: 
```

**Comment by matthewkeil** [comment]
> There is another, possibly related error I am seeing on my Lodestar node (v1.14.0-rc.1)
> 
> ```
> Jan-17 09:17:23.039[network]         error: Error setting agentVersion for the peer peerId=16Uiu2HAmKpgbkNA2AKYvt4qfdwFsSwuM7NwE8aMMHoMJiCBAxKAL - NotFound: 
> Error: NotFound: 
> ```

Talking with @tuyennhv this is a separate error.  I created issue #6365

**Comment by matthewkeil** [comment]
Fixed with #6366
